src/main/java/com/mineinabyss/geary/ecs/actions/SwitchToTargetAction.kt

```diff
@@ -0,0 +1,19 @@
+package com.mineinabyss.geary.ecs.actions
+
+import com.mineinabyss.geary.ecs.GearyEntity
+import com.mineinabyss.geary.ecs.components.Target
+import com.mineinabyss.geary.ecs.components.get
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+@SerialName("on_target")
+public class SwitchToTargetAction(
+    public val run: List<GearyAction>
+) : GearyAction() {
+    override fun runOn(entity: GearyEntity): Boolean {
+        val target = entity.get<Target>()?.entity ?: return false
+
+        return run.all{!it.runOn(target)}
```

Comment:
I think this should be `run.any { it.runOn(target) }`, we want to return true if *anything* happened when the action ran.

Reply:
Hm, both these implementations don't exactly do as we want as both `any` and `all` will stop iterating once the predicate is not matched. We need every action in `run` to execute their `runOn(target)`, and afterwards check if they all actions failed.

Essentially this, but could it be made nicer with kotlin magic? This feels like something I know but I'm blanking lol.
```
var temp = false
run.forEach {
    if(it.runOn(target)){
        temp = true
    }
}
return temp
```

Reply:
Oh my god you're right I remember doing this oversight at some point in the past 🤦 

It all comes back to haunt you

Reply:
What about `run.count { it.runOn(target) } != 0`?